confluent_kafka/avro/schema.py

```diff
@@ -0,0 +1,64 @@
+#!/usr/bin/env python
+#
+# Copyright 2018 Confluent Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+from avro.schema import PrimitiveSchema
+
+# Python 2 considers int an instance of str
+try:
+    string_type = basestring  # noqa
+except NameError:
+    string_type = str
+
+
+class GenericAvroRecord(dict):
+    """
```

Comment:
Consider composition instead of inheritance here because 1. it's a looser form of coupling (allows for more future flexibility), so is generally a better default 2. composition is used in other languages, e.g. https://github.com/confluentinc/avro/blob/master/lang/csharp/src/apache/main/Generic/GenericRecord.cs

Reply:
This is actually quite intentional as it allows me bind the schema with the record without needing to dereference the record contents in the serializer. Since all records are represented as dicts and the datum writer will only accept a dict I think this is the right move for python.

Reply:
i don't quite understand what you're trying to say, but there would be only minor difference in code between the two approaches (the composition approach would be a bit more but not much). Anyway, I'm ok with the inheritance relationship here.

Reply:
I guess I'm saying why call serialize(record.data) vs serialize(data). I'm not convinced I gain anything from composition here. On the contrary however Users can't treat this GenericAvroRecord exactly as if it were a dictionary. Just  like they could before I introduced a new type. i.e. record['stuff']='a' or record.get('stuff',None).

Reply:
Sure thing the other thing that is nice is your record types work exactly as they did in the past with no material changes. You can still use `record.put` and `record[key]` as you did previously since all records are dicts